### PR TITLE
[Bugfix] Console error in checkout and order confirmation page 

### PR DESCRIPTION
**Pull request by jcalcaben**
Fixes an intermittent bug that causes an error to be logged in the console when a visitor visits the checkout or order confirmation page for the first time.

Can be replicated by clearing the cache or using incognito mode.

This happens when the checkout or order confirmation dropin renders before the graphql endpoint has been set globally for the fetch-graphql package.

URL for testing:

- https://bugfix_null-endpoint2--boilerplate-commerce-dropins--jcalcaben.hlx.page/

### CODE REVIEW

File: blocks/commerce-checkout/commerce-checkout.js [hunk 1]
```diff
@@ -22,6 +24,12 @@ export default async function decorate(block) {
   // Initialize Drop-ins
   initializers.register(checkout.initialize, {});
 
+  // Set Fetch Endpoint (Service) if not yet set
+  const gqlConfig = checkout.getConfig();
+  if (!gqlConfig.endpoint) {
+    checkout.setEndpoint(await getConfigValue('commerce-core-endpoint'));
+  }
```
Comment (sirugh) [inline blocks/commerce-checkout/commerce-checkout.js:31]: This should already be handled in `loadEager` when we call `initializeDropins`.

Why is this necessary specifically in checkout/order confirmation pages? Can you give me some more info about the underlying bug?

**Review by sirugh — APPROVED** [review]
We should not need this, but it does fix the issue.

I think there's still a condition which will bite us in the future, and I'd like to understand more, but I don't want to hold up this fix.

@herzog31 when you return, perhaps you know why the session storage setting occurring within `config.js` is not actually awaiting. For example, initializeDropins is being called (setting endpoint) but before storage value is actually set in browser, the checkout block is running.

File: blocks/commerce-checkout/commerce-checkout.js (hunk 1 above)
Comment (sirugh) [reply blocks/commerce-checkout/commerce-checkout.js:31]: We spoke offline, but it appears that there is some condition wherein `getConfigValue` (which should _set_ the value if not present) which is called in the `initializeDropins` script, is not setting the config value before the `checkout` or `order confirmation` blocks are also calling the function, so there's something of a race condition still.

**Comment by dpatil-magento** [comment]
QA Approved.

**Comment by keoko** [comment]
@jcalcaben @sirugh It looks like that just waiting for `initializeDropins` in the `loadEager` function it may solve the issue:
```
await initializeDropins();
```
I am not sure about the impact of performance.

**Comment by sirugh** [comment]
Thanks @keoko - I'll open a PR to check.
